core/snmp/proxy-rpc-impl/src/main/java/org/opennms/netmgt/snmp/proxy/common/SnmpProxyRpcModule.java

```diff
@@ -176,6 +181,10 @@ private static final void addResult(SnmpResult result, String correlationId, Map
         final CompletableFuture<SnmpValue[]> future = SnmpUtils.getAsync(request.getAgent(), oids);
         return future.thenApply(values -> {
             final List<SnmpResult> results = new ArrayList<>(oids.length);
+            if (oids.length != values.length) {
```

Comment:
In what cases might this happen?

Reply:
I was thinking when values returned are less than number of oids requested.